main.py

```diff
@@ -111,13 +109,9 @@ def _get_user_quality_preference(self, stream_options):
             reversed(filtered_qualities)
         )
 
-    def setup_videoframe(self, stream_options, quality, coordinates):
-        """Sets ups a videoframe and with the provided stream information."""
-        videoframe = LiveVideoFrame(self.vlc_instance, stream_options, quality)
-        self.grid.addWidget(videoframe, coordinates.x, coordinates.y)
-        videoframe._move = self.move_frame
-
-        return videoframe
+    def setup_videoframe(self, stream_options, quality):
+        """Sets up a videoframe and with the provided stream information."""
+        self.grid.add_new_videoframe(self.vlc_instance, stream_options, quality, self.move_frame)
```

Comment:
IMO I'm not super stoked about the `self.move_frame` function being passed down to the individual videoframes since that feels a bit like backwards architecture (was so before the PR), but perhaps that's a thing for a later PR.